### ISSUE TITLE
Firebase 9 - No fromJSON method on AuthCredential

### ISSUE THREAD

**Issue by geoffgscott** [firebase/firebase-js-sdk#4926]
There does not seem to be be a way to create an AuthCredential from a JSON credential representation. The AuthCredential has a toJSON() method but not a fromJSON() method. Ideally a separate credentialFromJson() function could be provided that is separate from the AuthCredential class. 

 I see the EmailAuthCredential still has this method but I am unclear on the compatibility with a JSON representation that has been created from a standard AuthCredential. 

This is critical for handling third party auth provider in Electron workflows.

**Comment by looptheloop88** [comment]
You're correct, @geoffgscott. It seems that the AuthCredential class of JS SDK [version 9](https://firebase.google.com/docs/reference/js/v9/auth.authcredential) doesn't have the `fromJSON` method similar to [version 8](https://firebase.google.com/docs/reference/js/firebase.auth.AuthCredential).

Thanks for the feedback. I've tagged this issue as a feature request.

**Comment by sam-gc** [comment]
Hi, thanks for reaching out! `AuthCredential` is merely a base class that is never instantiated on its own. A `fromJSON` that works for all subclasses of `AuthCredential` would require the base class to import and reference all subclasses, which would prevent tree shaking.

Providers that do support recreation in this way _should_ have a `credentialFromJSON` method. It appears that during a refactor those got removed, so those need to be added back. These will look similar to [`OAuthProvider.credentialFromJSON`](https://firebase.google.com/docs/reference/js/v9/auth.oauthprovider.md#oauthprovidercredentialfromjson). That said, there won't be a generic `fromJSON` on `AuthCredential` itself.

**Comment by muryoh** [comment]
> Hi, thanks for reaching out! `AuthCredential` is merely a base class that is never instantiated on its own. A `fromJSON` that works for all subclasses of `AuthCredential` would require the base class to import and reference all subclasses, which would prevent tree shaking.

It makes a lot of sense for the modular version of the SDK, but shouldn't it have been kept on the compat version?
As far as I can tell, the `fromJSON` static function is still declared in the typings but no longer available at runtime

**Comment by boldtrn** [comment]
Just wanted to +1 this, as this is breaking external implementations like firebaseui-web.

**Comment by yurist38** [comment]
Hello. Any progress on this one? I've noticed issues with linking accounts using the `firebase-ui` lib. Would be great to get it fixed...
